### PR TITLE
Incremental docs publish, travis builds n more enhancements

### PR DESCRIPTION
**Pull request by sivakumar-kailasam**
This PR introduces the following goodness,

- Travis scripts that publish incremental docs. Modified the module to skip processing any version of docs that already exists in the target DB
- Ability to run the module for a specific version of docs that can be passed as an arg. Eg `node index.js 2.10`
- YUIdocs were fetched twice from S3, the redundant call is removed
- Generate log msgs so that travis doesn't timeout
- Increase timeout between requests so that requests don't fail
- Removed unused deps, updated outdated ones & added npm script to fix file formatting using standard cli
- Added yarn.lock & editorconfig file

**_Things that requires travis org admin access,_**
-  Enable travis for this repo
- Set the following,
```shell
travis env set AWS_ACCESS_KEY_ID "$AWS_ACCESS_KEY_ID"
travis env set AWS_SECRET_ACCESS_KEY "$AWS_SECRET_ACCESS_KEY"
travis env set COUCH_USERNAME "$COUCH_USERNAME"
travis env set COUCH_PASSWORD "$COUCH_PASSWORD"
```
- Replace the **COUCH_URL** in _.travis.yml_ with actual db
- If the target couch db is a new one, we need to populate it with at least _v1_ docs since generating docs for both v1.x and v2.x takes more than 50mins which is the max period for travis builds.

Addresses ember-learn/ember-api-docs#66

### CODE REVIEW

**Comment by sivakumar-kailasam** [comment]
An additional step of flushing travis in heroku is to be added to the `.travis.yml`. Will raise it in another PR.

File: lib/fetch.js [hunk 1]
```diff
@@ -84,6 +84,58 @@ function filterReleaseDocs (document) {
   return versionRegex.test(tag) && /-docs\.json/.test(key)
 }
 
-module.exports = function fetch () {
-  return getObjects().then(downloadReleaseDocs)
+function getNewDocsToIndex (db, docs) {
```
Comment (Gaurav0) [inline lib/fetch.js]: possible to move this to separate file?

**Review by Gaurav0 — APPROVED** [review]
Looks awesome! Thanks so much!

File: lib/fetch.js (hunk 1 above)
Comment (sivakumar-kailasam) [reply lib/fetch.js]: Sure

**Pull request merged**